src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/service/MappingService.kt

```diff
@@ -70,27 +70,32 @@ class MappingService(
 
   suspend fun deleteVisitMappings() = visitIdRepository.deleteAll()
 
-  fun getVisitMappingsByMigrationId(pageRequest: Pageable, migrationId: String): Mono<Page<MappingDto>> {
-    return visitIdReactiveRepository.findAllByLabelAndMappingTypeOrderByLabelDesc(
-      label = migrationId,
-      MappingType.MIGRATED,
-      pageRequest
-    ).collectList()
-      .zipWith(visitIdReactiveRepository.countAllByLabelAndMappingType(migrationId, mappingType = MappingType.MIGRATED))
-      .map { t ->
-        PageImpl(
-          t.t1.map { mapping ->
-            MappingDto(
-              mapping.nomisId,
-              mapping.vsipId,
-              mapping.label,
-              mapping.mappingType.name
-            )
-          },
-          pageRequest, t.t2
+  suspend fun getVisitMappingsByMigrationId(pageRequest: Pageable, migrationId: String): Page<MappingDto> =
+    coroutineScope {
+      val visits = async {
```

Comment:
this way has the added advantage that the `find` and `count` can happen in parallel

Reply:
Nice